components/dada-parse/src/parser/code.rs

```diff
@@ -312,6 +312,15 @@ impl CodeParser<'_, '_> {
         } else if let Some((id_span, id)) = self.eat(Identifier) {
             tracing::debug!("identifier");
             Some(self.add(ExprData::Id(id), id_span))
+        } else if let Some((return_span, _)) = self.eat(Keyword::Return) {
```

Comment:
I'm debating if `parse_expr_0` is the right place for this. I think I would put it at the highest level (e.g., `parse_expr` itself).

As an example of why, consider this (and we should make this into a test)

```
fn foo() {
    return * 3
}
```

I think I want that to be a parse error, but I believe it will parse with the setup you have here. The same is true for any binary operator that cannot begin an expression (e.g., `return && true`).